src/main/java/org/folio/rest/impl/LoginAPI.java

```diff
@@ -199,31 +220,228 @@ private Future<String> fetchToken(JsonObject payload, String tenant,
+
+      return response.bodyAsJsonObject();
     });
   }
 
-  private Future<String> fetchRefreshToken(String userId, String sub, String tenant,
-    String okapiURL, String requestToken) {
-    HttpRequest<Buffer> request = WebClientFactory.getWebClient(vertx).postAbs(okapiURL + "/refreshtoken");
+  private Future<HttpResponse<Buffer>> logout(String tenant,
+      String okapiURL, String accessToken, String refreshToken) {
+    HttpRequest<Buffer> request = WebClientFactory.getWebClient(vertx).postAbs(okapiURL + TOKEN_LOGOUT_ENDPOINT);
+
     request.putHeader(XOkapiHeaders.TENANT, tenant)
-      .putHeader(XOkapiHeaders.TOKEN, requestToken);
+      .putHeader(XOkapiHeaders.TOKEN, accessToken);
 
-    JsonObject payload = new JsonObject().put("userId", userId).put("sub", sub);
-    return request
-      .expect(ResponsePredicate.SC_CREATED)
-      .expect(ResponsePredicate.JSON)
-      .sendJsonObject(payload)
-      .map(response -> response.bodyAsJsonObject().getString("refreshToken")
-      );
+      return request.sendJson(new JsonObject().put(REFRESH_TOKEN, refreshToken)).map(response -> {
+        return response;
+      });
+  }
+
+  private Future<HttpResponse<Buffer>> logoutAll(String tenant,
+      String okapiURL, String accessToken) {
+    HttpRequest<Buffer> request = WebClientFactory.getWebClient(vertx).postAbs(okapiURL + TOKEN_LOGOUT_ALL_ENDPOINT);
+
+    request.putHeader(XOkapiHeaders.TENANT, tenant)
+      .putHeader(XOkapiHeaders.TOKEN, accessToken);
+
+    return request.send();
+  }
+
+  private Future<HttpResponse<Buffer>> fetchRefreshToken(String tenant,
+      String okapiURL, String accessToken, String refreshToken) {
+    HttpRequest<Buffer> request = WebClientFactory.getWebClient(vertx).postAbs(okapiURL + TOKEN_REFRESH_ENDPOINT);
+
+    request.putHeader(XOkapiHeaders.TENANT, tenant)
+      .putHeader(XOkapiHeaders.TOKEN, accessToken);
+
+    return request.sendJson(new JsonObject().put(REFRESH_TOKEN, refreshToken));
+  }
+
+  private boolean usesTokenSignLegacy(String endpoint) {
+    return endpoint.equals(TOKEN_SIGN_ENDPOINT_LEGACY);
+  }
+
+  // For documentation of the cookie attributes please see:
+  // https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Set-Cookie
+  private String refreshTokenCookie(String refreshToken, String refreshTokenExpiration) {
+    // The refresh token expiration is the time after which the token will be considered expired.
+    var exp = Instant.parse(refreshTokenExpiration).getEpochSecond();
+    var ttlSeconds = exp - Instant.now().getEpochSecond();
+    var sb = new StringBuilder("refreshToken=");
+    sb.append(refreshToken);
+
+    // Path indicates the path that must exist in the requested URL for the client to send the
+    // Cookie header.
+    sb.append("; HttpOnly; Secure; Path=/authn; ");
+
+    // The Max-Age attribute is the number seconds until the token and cookie expires. It also ensures
+    // that the cookie will persist between browser tab sessions.
```

Comment:
The first sentence is already explained in https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Set-Cookie and should not be repeated here.

The second sentence is contrary to the document and should be reworded: We use Max-Age and not Expires to make it a session cookie that the browser deletes at the end of a browser session.

Reply:
Removed comment.